File: tests/pages/firefox/whatsnew/whatsnew.py
# This Source Code Form is subject to the terms of the Mozilla Public
# License, v. 2.0. If a copy of the MPL was not distributed with this
# file, You can obtain one at https://mozilla.org/MPL/2.0/.

from selenium.webdriver.common.by import By

from pages.base import BasePage
from pages.regions.send_to_device import SendToDevice


class FirefoxWhatsNewPage(BasePage):
    _URL_TEMPLATE = "/{locale}/firefox/whatsnew/"

    _qr_code_locator = (By.CSS_SELECTOR, ".qr-code img")

    @property
    def send_to_device(self):
        return SendToDevice(self)

    @property
    def is_qr_code_displayed(self):
        return self.is_element_displayed(*self._qr_code_locator)

File: media/js/firefox/whatsnew/whatsnew.js
/*
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 */

// create namespace
if (typeof window.Mozilla === 'undefined') {
    window.Mozilla = {};
}

(function () {
    'use strict';

    var sendTo = document.getElementById('send-to-device');

    if (sendTo) {
        var form = new window.Mozilla.SendToDevice();
        form.init();
    }
})();
